nxdrive/gui/folders_dialog.py

```diff
@@ -388,9 +389,28 @@ def _add_subgroup_new_folder(self, layout: QHBoxLayout, /) -> None:
         layout.addWidget(self.new_folder)
         layout.addWidget(self.new_folder_button)
 
+    def _find_folders_duplicates(self) -> List[str]:
+        if bool(self.new_folder.text()):
+            return []
+        all_paths = self.paths.keys()
+        uploaded_folders = [
+            path.name
+            for path in all_paths
+            if path.is_dir() and path.parent not in all_paths
```

Comment:
Reverse the conditions to prevent OS access for each file. You could also merge the last condition to prevent having 2 comprehension lists.

Define `parent = self.remote_folder_ref` before the list and then:
```suggestion
            if (
                path.parent not in all_paths
                and path.is_dir()
                and self.engine.remote.exists_in_parent(parent, folder)
            )
```